Make MongoDB and Redis datastore plugins optional

arkamar opened the pull request:
This commit introduces the `ENABLE_DS_MONGO` and `ENABLE_DS_REDIS` options, allowing users to explicitly enable or disable the MongoDB and Redis datastore plugins. Without these options, the plugins had automagic properties, being included whenever their dependencies were found, which might not be intentional.

This change is especially useful for downstream packagers, as it allows them to control the dependencies explicitly and avoid unintended linkage. It reduces unnecessary dependencies in environments where these plugins are not needed.

michalvasko reviewed (changes requested):
No problem and I think I would even go further by making the default values `OFF`. If the user is not using these plugins and they are enabled, it causes issues so I think having to explicitly enable them makes sense. Also, please change the target branch to `devel`, we do not accept PRs into `master`.

arkamar commented:
Yeah, I agree, `OFF` is better but I wasn't sure what is preferred default, so I made it in a way to preserve original behavior. Anyway, I changed it to `OFF`, rebased to `devel` and changed the target branch.

michalvasko commented:
Right, the CI needs to be updated, can you do it?

arkamar commented:
Thanks!
